client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -118,6 +122,48 @@ static Request index(IndexRequest indexRequest) {
         return new Request(method, endpoint, parameters.getParams(), entity);
     }
 
+    static Request update(UpdateRequest updateRequest) throws IOException {
```

Comment:
Well, I'm on the fence on this. We already use CheckedFunction for response parsers I think we could also use CheckedFunction for request converters? I don't feel like it adds a lot of boling code.